pkg/util/rankedset/rankedset.go

```diff
@@ -0,0 +1,162 @@
+package rankedset
+
+import "github.com/google/btree"
+
+// Item represents a single object in a RankedSet.
+type Item interface {
+	// Key returns the unique identifier for this item.
+	Key() string
+	// Rank is used to sort items.
+	// Items with the same rank are sorted lexicographically based on Key.
+	Rank() int64
+}
+
+// RankedSet stores Items based on Key (uniqueness) and Rank (sorting).
+type RankedSet struct {
+	rank *btree.BTree
+	set  map[string]*treeItem
```

Comment:
Why creating artificial `treeItem` instead of just using `Item` here? Especially that the only thing that `treeItem` has is `Item`?

Reply:
It prevents me from having to cast it back and fort.  I have to perform operations on both the tree and the map with the same data, and it just makes it simpler if they hold the same structure.